### PR TITLE
New Asexual pride flag

### PR DESCRIPTION
**Pull request by eepy-goddess**
I added in the asexual pride flag. Asexuals can now happily show their pride while also showing their neofetches.

### CODE REVIEW

**Review by eepy-goddess — COMMENTED** [review]
I have followed all your your naming conventions (I hope) and I reviewed and tested all added code.

File: main.c [hunk 1]
```diff
@@ -373,6 +380,28 @@ static void get_color(flag_type_t flag_type, float offset, float theta, uint8_t
...
+	{ /* white to purple */
+	    theta = 1 - ((theta - FLAG_4_2) / (FLAG_4_3 - FLAG_4_2));
+	    gen_color(ASE_WHI, ASE_PUR, theta, ASE_FCT, red, green, blue);
+	}
+	else if (FLAG_4_3 <= theta && FLAG_4_4 > theta)
+	{ /* purple back to black */
+	    theta = 1 - ((theta - FLAG_4_3) / (FLAG_4_4 - FLAG_4_3));
+	    gen_color(ASE_PUR, ASE_BLA, theta, ASE_FCT, red, green, blue);
+	}
+	break;
```
Comment (Elsa002) [inline main.c]: The indentation here is different, also replace tabs with spaces

File: main.c [hunk 2]
```diff
@@ -412,6 +441,10 @@ static size_t get_codes_len(flag_type_t flag_type)
     case FLAG_TYPE_GENDERFLUID:
         return ARRAY_SIZE(codes_gfl);
         break;
+    case FLAG_TYPE_ASEXUAL:
```
Comment (Elsa002) [inline main.c]: Add empty line before for consistency, also indentations and replace tabs with spaces

File: main.c [hunk 3]
```diff
@@ -451,6 +484,10 @@ static const char* get_codes(flag_type_t flag_type)
     case FLAG_TYPE_GENDERFLUID:
         return codes_gfl;
         break;
+   case FLAG_TYPE_ASEXUAL:
```
Comment (Elsa002) [inline main.c]: Add empty line before for consistency, also indentations and replace tabs with spaces

File: main.c (hunk 1 above)
Comment (eepy-goddess) [reply main.c]: I fixed the indentations and replaced the tabs with spaces

**Pull request merged**